### PR TITLE
Issue #374 "params" should be used on overrides

### LINKED ISSUE #374: "params" should be used on overrides
```
class Base
{
  public virtual void Method(params int[] numbers)
  {
    ...
  }
}
class Derived : Base
{
  public override void Method(int[] numbers) // Noncompliant, the params is missing.
  {
    ...
  }
}
```
base class has overridden/removed the `params` behavior. Now must pass explicit array.
major
suspicious

R# ID: BaseMemberHasParams

https://jira.sonarsource.com/browse/RSPEC-3262

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #374

### CODE REVIEW

File: src/Tests/SonarLint.UnitTest/TestCases/MethodOverrideNoParams.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        }
+    }
+
+    class Derived2 : Base
+    {
+        public override void Method(params int[] numbers)
+        {
+        }
+    }
+}
```
Comment (duncanpMS) [inline src/Tests/SonarLint.UnitTest/TestCases/MethodOverrideNoParams.cs]: Potential additional test case: abstract method

File: src/Tests/SonarLint.UnitTest/TestCases/MethodOverrideNoParams.Fixed.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    class Derived : Base
+    {
+        public override void Method(params int[] numbers) // Noncompliant, the params is missing.
+        {
+        }
+    }
+
+    class Derived2 : Base
+    {
+        public override void Method(params int[] numbers)
```
Comment (duncanpMS) [inline src/Tests/SonarLint.UnitTest/TestCases/MethodOverrideNoParams.Fixed.cs]: If the arguments are indented on a separate line from the method name, the indentation is lost after applying the fix.
It looks a bit odd, but I guess that is how the editor thinks the doc should be edited (the formatting doesn't change if you select the whole doc and select Edit, Advanced, Format Document.

**Comment by duncanpMS** [comment]
LGTM

File: src/Tests/SonarLint.UnitTest/TestCases/MethodOverrideNoParams.Fixed.cs (hunk 2 above)
Comment (tvsonar) [inline src/Tests/SonarLint.UnitTest/TestCases/MethodOverrideNoParams.Fixed.cs]: Thanks, I fixed this by adding back the trivia from the old parameter to the new one.

**Pull request merged**